lib/providers/torrentleech.js

```diff
@@ -0,0 +1,88 @@
+const Promise = require('bluebird');
+const request = Promise.promisify(require('request'));
+const TorrentProvider = require('../TorrentProvider');
+
+class TorrentLeech extends TorrentProvider {
+	constructor() {
+		super({
+			name: 'TorrentLeech',
+			requireAuthentification: true,
+			supportCookiesAuthentification: true,
+			supportCredentialsAuthentification: false,
+			baseUrl: 'https://www.torrentleech.org',
+			searchUrl:
+				'/torrents/browse/list/query/{query}/categories/{cat}/orderby/seeders/order/desc',
+			categories: {
+				All:
+					'',
+				Movies: '8,9,11,37,43,14,12,13,47,15,29',
+				TV: '26,32,27',
+				Games: '17,42,18,19,40,20,21,39,22,28,30,48',
+				Apps: '23,24,25,33',
+				Education: '38',
+				Animation: '34,35',
+				Books: '45,46',
+				Music: '31,16',
+				Foreign: '36,44'
+			},
+			defaultCategory: 'All',
+		});
+	}
+
+	search(query, category) {
+		const url = this.getUrl(category, query);
+		if (url === null) {
+			return undefined;
+		}
+		return request({ url, jar: this.cookieJar, headers: { 'User-Agent': 'curl/7.37.0' } })
+			.then(response => {
+				const results = JSON.parse(response.body).torrentList;
+				return (
+					results &&
+					results.map(r => ({
+						provider: this.name,
+						title: r.name,
+						time: r.addedTimestamp,
+						seeds: r.seeders,
+						peers: r.leechers,
+						size: this.humanFileSize(r.size),
+						filename: r.filename,
+						fid: r.fid,
+
+						rating: r.rating,
+						categoryID: r.categoryID,
+						new: r.new,
+						numComments: r.numComments,
+						tags: r.tags,
+						imdbID: r.imdbID,
+						igdbID: r.igdbID,
+						tvmazeID: r.tvmazeID
+					}))
+				);
+			});
+	}
+
+	getTorrentDetails(torrent) {
+		throw new Error('Not implemented');
+	}
+
+	downloadTorrent() {
+		throw new Error('Not implemented');
```

Comment:
Is it difficult to implement those methods ?

Reply:
I can implement the `downloadTorrent` method, but concerning `getTorrentDetails` i a bit confused what should be return if any. If i should return the Html, it is a bit funky, to extract using json, but give the user the raw html, also torrentleech have a long information tab for a torrent, should i give the entire tab info, or just part of it ? and if part of it, then what should be must that ?
I haven't seen any example in previous providers so i wasn't sure.
If you want i can implement it as i do best.
Waiting for your reply.

Reply:
Sorry for the delay, I have been working a lot lately.

For the "getTorrentDetails" method, the idea is to get whatever information there is about the torrent. So just do your best ;)

Thanks.